docs/content/docs/dev/datastream/fault-tolerance/state.md

```diff
@@ -474,9 +474,10 @@ ttl_config = StateTtlConfig \
 For more fine-grained control over some special cleanup in background, you can configure it separately as described below.
 Currently, heap state backend relies on incremental cleanup and RocksDB backend uses compaction filter for background cleanup.
 
-##### Cleanup in full snapshot
+##### Cleanup in full scan snapshot
 
-Additionally, you can activate the cleanup at the moment of taking the full state snapshot which 
+Additionally, you can activate the cleanup at the moment of taking the full scan state snapshot (including the canonical savepoint,
+or the full/incremental checkpoint of hashmap state-backend, or the full checkpoint of RocksDB state-backend), which 
```

Comment:
IIUC, the intention is to list all the cases except RocksDB native savepoints and incremental checkpoints, right? Why not to do that explicitly?
 
Listing all state backends and their modes here might require changes in the future.

Reply:
I think telling users with a direct explanation of what this API can take effect could be more accurate. And I believe these descriptions would change if we apply new implementations for different state-backend, e.g FLINK-28699 or FLIP-151.

Reply:
Then how about listing the supported combinations explicitly under "supported"?
```
Additionally, you can activate the cleanup at the moment of taking the full scan state snapshot. 
This is only supported for canonical savepoints and full checkpoints with HashMap and RocksDB state backends.
```
From the current wording (`including ...`) it's not clear to me whether the list is exhaustive or not.

Reply:
Got it, I will refactor these descriptions.